asistente_ladm_col/logic/quality/line_quality_rules.py

```diff
@@ -0,0 +1,731 @@
+        inner_rings_layer = processing.run("native:addautoincrementalfield",
+                                           {'INPUT': tmp_inner_rings_layer,
+                                            'FIELD_NAME': 'AUTO',
+                                            'START': 0,
+                                            'GROUP_FIELDS': [],
+                                            'SORT_EXPRESSION': '',
+                                            'SORT_ASCENDING': True,
+                                            'SORT_NULLS_FIRST': False,
+                                            'OUTPUT': 'memory:'})['OUTPUT']
+
+        id_field_idx = inner_rings_layer.fields().indexFromName(id_field)
+        auto_idx = inner_rings_layer.fields().indexFromName('AUTO')
+        request = QgsFeatureRequest().setSubsetOfAttributes([id_field_idx, auto_idx])
+        dict_inner_rings = {'{}-{}'.format(feature[id_field], feature['AUTO']): feature for feature in inner_rings_layer.getFeatures(request)}
+
+        # spatial joins between boundary and inner rings
+        # we use as input layer the rings because it is the existing information
+        # in the terrain polygons and filters better because they are less records
+        spatial_join_inner_rings_boundary_layer = processing.run("qgis:joinattributesbylocation",
+                                                                 {'INPUT': boundary_layer,
+                                                                  'JOIN': inner_rings_layer,
+                                                                  'PREDICATE': [0],  # Intersects
+                                                                  'JOIN_FIELDS': [id_field, 'AUTO'],
+                                                                  'METHOD': 0,
+                                                                  'DISCARD_NONMATCHING': True,
+                                                                  'PREFIX': '',
+                                                                  'OUTPUT': 'memory:'})['OUTPUT']
+
+        # The id field has the same name for both layers
+        # This list is only used to check plot's inner rings without boundaries
+        list_spatial_join_boundary_inner_rings = [{'plot_ring_id': '{}-{}'.format(feature[id_field + '_2'], feature['AUTO']), 'boundary_id': feature[id_field]}
+                                                  for feature in spatial_join_inner_rings_boundary_layer.getFeatures()]
+
+        # list create for filter inner rings from spatial join with between plot and boundary
+        list_spatial_join_boundary_plot_ring = [{'plot_id': feature[id_field + '_2'],
+                                                 'boundary_id': feature[id_field]}
+                                                for feature in spatial_join_inner_rings_boundary_layer.getFeatures()]
```

Comment:
Avoid several calls to `spatial_join_inner_rings_boundary_layer.getFeatures()`. It can be done in a single for, settings those two lists.
(TODO: for a follow-up PR)

Reply:
Thanks, resolve in dc0ba8d